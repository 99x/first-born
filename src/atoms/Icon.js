import { Platform } from "react-native";
import React, { Component } from "react";
import Ionicon from "react-native-vector-icons/Ionicons";
import PropTypes from "prop-types";
import { commonColors } from "../utils/color";

export class Icon extends Component {
    render() {
        const { name, size, color, ...otherProps } = this.props;

        if (!name.includes("logo")) {
            return (
                <Ionicon
                    name={
                        Platform.OS === "android" ? "md-" + name : "ios-" + name
                    }
                    size={size}
                    color={color}
                    {...otherProps}
                />
            );
        }
        return (
            <Ionicon name={name} size={size} color={color} {...otherProps} />
        );
    }
}

Icon.propTypes = {
    ...Ionicon.propTypes,
    name: PropTypes.string
};

Icon.defaultProps = {
    color: commonColors.white,
    size: 18
};
