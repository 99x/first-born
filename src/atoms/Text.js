import React, { Component } from "react";
import { Text } from "react-native";
import PropTypes from 'prop-types';
import { commonColors } from "../utils/color";
import { getFontSize } from "../variables/textSizeVariables";

class TextAE extends Component {

    render() {
        const { color, size, bold, children, ...otherProps } = this.props;

        const textColor = !color ? commonColors.black : color;

        const textSize = !size ? "p" : size;

        const textBold = !bold ? "normal" : "bold";

        const textStyle = { color: textColor, fontSize: getFontSize(textSize), fontWeight: textBold };

        return (
            <Text
                ref={c => (this._root = c)}
                style={textStyle}
                {...otherProps}
            >
                {children}
            </Text>
        )
    }
}

TextAE.propTypes = {
    bold: PropTypes.bool,
    color: PropTypes.string,
    size: PropTypes.oneOf(["h1", "h2", "h3", "h4", "h5", "h6", "p", "callout", "sub_heading", "footnote", "caption_big", "caption_small"]),
    ...Text.propTypes
};

TextAE.defaultProps = {
    size: "p",
    bold: false,
    color: commonColors.black
};

export { TextAE as Text };
