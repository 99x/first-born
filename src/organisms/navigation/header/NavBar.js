import React, { Component } from "react";
import { View, StatusBar, Platform } from "react-native";
import PropTypes from "prop-types";
import { deviceVariables } from "../../../variables/deviceVariables";
import { commonColors, shadeColor } from "../../../utils/color";
import { isIphoneX } from "../../../utils/platform";

export class NavBar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            orientation: "portrait"
        };
    }

    layoutChange(val) {
        let maxComp = Math.max(deviceVariables.width, deviceVariables.height);
        if (val.width >= maxComp) {
            this.setState({ orientation: "landscape" });
        } else {
            this.setState({ orientation: "portrait" });
        }
    }

    calculateHeight(mode, inset) {
        const { toolbarHeight } = this.props;

        const insetValues =
            mode === "portrait" ? inset.portrait : inset.landscape;

        let oldHeight = null;

        if (toolbarHeight != undefined) {
            oldHeight = toolbarHeight;
        }

        let height = oldHeight + insetValues.bottomInset;

        if (!isIphoneX()) {
            height -= 20;
        }

        return height;
    }

    calculatePadder(mode, inset) {
        const { paddingTop } = this.props;
        const insetValues =
            mode === "portrait" ? inset.portrait : inset.landscape;
        const topPadder = paddingTop + insetValues.topInset;
        return topPadder;
    }

    render() {
        const {
            statusBarColor,
            statusBarContent,
            transparent,
            translucent,
            children,
            toolbarHeight,
            paddingTop,
            style,
            shadowProperties,
            ...otherProps
        } = this.props;
        const { orientation } = this.state;

        let navBarStyle = { ...style };

        if (!transparent) {
            navBarStyle = { ...navBarStyle, ...shadowProperties };
        }

        if (Platform.OS === "ios") {
            if (isIphoneX()) {
                return (
                    <View
                        onLayout={e => this.layoutChange(e.nativeEvent.layout)}
                        style={{
                            height: this.calculateHeight(
                                orientation,
                                deviceVariables.Inset
                            ),
                            paddingTop: this.calculatePadder(
                                orientation,
                                deviceVariables.Inset
                            ),
                            backgroundColor: statusBarColor
                        }}
                    >
                        <StatusBar
                            backgroundColor={shadeColor(statusBarColor, 0.2)}
                            barStyle={statusBarContent}
                            translucent={transparent ? true : translucent}
                        />

                        <View
                            ref={c => (this._root = c)}
                            {...otherProps}
                            style={navBarStyle}
                        >
                            {children && children}
                        </View>
                    </View>
                );
            }
            return (
                <View
                    onLayout={e => this.layoutChange(e.nativeEvent.layout)}
                    style={{
                        height: this.calculateHeight(
                            orientation,
                            deviceVariables.Inset
                        ),
                        paddingTop: paddingTop,
                        backgroundColor: statusBarColor
                    }}
                >
                    <StatusBar
                        backgroundColor={shadeColor(statusBarColor, 0.2)}
                        barStyle={statusBarContent}
                        translucent={transparent ? true : translucent}
                    />

                    <View
                        ref={c => (this._root = c)}
                        {...otherProps}
                        style={navBarStyle}
                    >
                        {children && children}
                    </View>
                </View>
            );
        }

        return (
            <View
                onLayout={e => this.layoutChange(e.nativeEvent.layout)}
                style={{
                    height: toolbarHeight,
                    paddingTop: paddingTop,
                    backgroundColor: statusBarColor
                }}
            >
                <StatusBar
                    backgroundColor={shadeColor(statusBarColor, 30)}
                    barStyle={statusBarContent}
                    translucent={transparent ? true : translucent}
                />

                <View
                    ref={c => (this._root = c)}
                    {...otherProps}
                    style={navBarStyle}
                >
                    {children && children}
                </View>
            </View>
        );
    }
}

NavBar.propTypes = {
    transparent: PropTypes.bool,
    statusBarColor: PropTypes.string,
    statusBarContent: PropTypes.oneOf(["light-content", "dark-content"]),
    ...View.propTypes
};

NavBar.defaultProps = {
    statusBarColor:
        Platform.OS === "android" ? commonColors.primary : "#f8f8f8",
    toolbarHeight: Platform.OS === "android" ? 56 : 64,
    paddingTop: Platform.OS === "android" ? 0 : 20,
    statusBarContent:
        Platform.OS === "android" ? "light-content" : "dark-content",
    style: {
        flex: 1,
        flexDirection: "row"
    },
    shadowProperties: {
        elevation: 3,
        shadowColor: commonColors.black,
        shadowOffset: { width: 0, height: 3 },
        shadowOpacity: 0.4,
        shadowRadius: 1.2
    }
};
