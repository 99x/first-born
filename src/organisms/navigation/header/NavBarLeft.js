import React, { Component } from "react";
import { View, StyleSheet } from "react-native";

export class NavBarLeft extends Component {
    render() {
        const { children, ...otherProps } = this.props;

        return (
            <View style={styles.container} {...otherProps}>
                {children && children}
            </View>
        );
    }
}

NavBarLeft.propTypes = {};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "center"
    }
});
