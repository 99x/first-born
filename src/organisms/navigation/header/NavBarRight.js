import React, { Component } from "react";
import { View, StyleSheet } from "react-native";

export class NavBarRight extends Component {

    render() {

        const { children, ...otherProps } = this.props;

        return (
            <View style={styles.container} {...otherProps} >
                {children && children}
            </View>
        );
    }
}

NavBarRight.propTypes = {
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "center",
    }
})