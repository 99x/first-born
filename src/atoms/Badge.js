import React, { Component } from "react";
import { View, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Badge extends Component {
    render() {
        const {
            color,
            children,
            ...otherProps
        } = this.props;

        const badgeColor = color;

        let badgeStyle = [styles.defaultBadge, {
            borderRadius: 15,
            backgroundColor: badgeColor
        }];

        let textStyle = {
            color: commonColors.white
        };

        let iconColor = commonColors.white;;

        const newChildren = React.Children.map(children, child =>
            child && child.type === Text
                ? React.cloneElement(child, {
                    ...child.props,
                    style: { ...child.props.style, ...textStyle },
                    size: "footnote"
                })
                : child && child.type === Icon
                    ? React.cloneElement(child, {
                        ...child.props,
                        color: iconColor,
                        size: 15
                    })
                    : child && child.type === Image
                        ? React.cloneElement(child, {
                            ...child.props,
                            style: {
                                ...child.props.style,
                                width: 15,
                                height: 15
                            }
                        })
                        : null
        );

        return (
            <View
                ref={c => (this._root = c)}
                style={badgeStyle}
                {...otherProps}
            >
                {newChildren}
            </View>
        );
    }
}

Badge.propTypes = {
    color: PropTypes.string,
    ...View.propTypes
};

Badge.defaultProps = {
    color: commonColors.primary,
};

const styles = StyleSheet.create({
    defaultBadge: {
        margin: 5,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        paddingHorizontal: 10,
        paddingVertical: 5
    },
});
