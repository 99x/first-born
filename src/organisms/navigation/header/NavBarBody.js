import React, { Component } from "react";
import { View, StyleSheet, Platform } from "react-native";
import PropTypes from "prop-types";
import { Text } from "../../../atoms/Text";
import { commonColors } from "../../../utils/color";

export class NavBarBody extends Component {
    render() {
        const { style, children, color, ...otherProps } = this.props;

        const newChildren = React.Children.map(children, child =>
            child && child.type === Text
                ? React.cloneElement(child, {
                      ...child.props,
                      style: { ...child.props.style },
                      size: "h6",
                      bold: true,
                      color
                  })
                : null
        );

        return (
            <View style={styles.container} {...otherProps}>
                {newChildren && newChildren}
            </View>
        );
    }
}

NavBarBody.propTypes = {
    color: PropTypes.string,
    ...View.propTypes
};

NavBarBody.defaultProps = {
    color: Platform.OS === "android" ? commonColors.white : commonColors.black
};

const styles = StyleSheet.create({
    container: {
        flex: 3,
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "center"
    }
});
