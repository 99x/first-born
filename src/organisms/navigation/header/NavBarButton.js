import React, { Component } from "react";
import { Platform, Image, TouchableOpacity } from "react-native";
import PropTypes from "prop-types";
import { Button } from "../../../atoms/Button";
import { Icon } from "../../../atoms/Icon";
import { Text } from "../../../atoms/Text";

export class NavBarButton extends Component {

    render() {
        const { type, children, ...otherProps } = this.props;

        let newChildren = []

        if (children) {

            newChildren = React.Children.map(children, child => child && child.type === Text ?
                React.cloneElement(child, { size: "p", ...child.props }) : child && child.type === Icon ?
                    React.cloneElement(child, { size: 25, ...child.props }) : child && child.type === Image ?
                        React.cloneElement(child, { style: { width: 25, height: 25 }, ...child.props }) : null);

        } else if (type) {

            switch (type) {
                case "back":
                    newChildren.push(<Icon name="arrow-back" size={25} />);
                    break;
                case "search":
                    newChildren.push(<Icon name="search" size={25} />);
                    break;
                case "drawer":
                    newChildren.push(<Icon name="menu" size={25} />);
                    break;
            }

            if (Platform.OS === "ios" && type === "back") {
                newChildren.push(<Text>Back</Text>);
            }

        }

        if (Platform.OS === "android") {
            return (
                <Button {...otherProps} transparent>
                    {newChildren && newChildren}
                </Button>
            );
        } else {
            return (
                <Button {...otherProps} outline transparent color="#0a60ff">
                    {newChildren && newChildren}
                </Button>
            );
        }
    }
}

NavBarButton.propTypes = {
    type: PropTypes.oneOf(["back", "search", "drawer"]),
    ...TouchableOpacity.propTypes
}

NavBarButton.defaultProps = {
    type: undefined
}
