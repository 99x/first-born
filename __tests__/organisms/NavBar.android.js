import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Text } from "../../src/atoms/Text";
import { Icon } from "../../src/atoms/Icon";
import { NavBar } from "../../src/organisms/navigation/header/NavBar";
import { NavBarLeft } from "../../src/organisms/navigation/header/NavBarLeft";
import { NavBarRight } from "../../src/organisms/navigation/header/NavBarRight";
import { NavBarBody } from "../../src/organisms/navigation/header/NavBarBody";
import { NavBarButton } from "../../src/organisms/navigation/header/NavBarButton";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "android";
    return Platform;
});

it("renders NavBar with title", () => {
    const tree = renderer
        .create(
            <NavBar>
                <NavBarLeft />
                <NavBarBody>
                    <Text>Title</Text>
                </NavBarBody>
                <NavBarRight />
            </NavBar>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders NavBar with back button", () => {
    const tree = renderer
        .create(
            <NavBar>
                <NavBarLeft>
                    <NavBarButton type="back" />
                </NavBarLeft>
                <NavBarBody>
                    <Text>Title</Text>
                </NavBarBody>
                <NavBarRight />
            </NavBar>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders NavBar with drawer button", () => {
    const tree = renderer
        .create(
            <NavBar>
                <NavBarLeft>
                    <NavBarButton type="drawer" />
                </NavBarLeft>
                <NavBarBody>
                    <Text>Title</Text>
                </NavBarBody>
                <NavBarRight />
            </NavBar>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders NavBar with search button", () => {
    const tree = renderer
        .create(
            <NavBar>
                <NavBarLeft>
                    <NavBarButton type="search" />
                </NavBarLeft>
                <NavBarBody>
                    <Text>Title</Text>
                </NavBarBody>
                <NavBarRight />
            </NavBar>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders NavBar with custom button", () => {
    const tree = renderer
        .create(
            <NavBar>
                <NavBarLeft />
                <NavBarBody>
                    <Text>Title</Text>
                </NavBarBody>
                <NavBarRight>
                    <NavBarButton>
                        <Icon name="heart" />
                    </NavBarButton>
                </NavBarRight>
            </NavBar>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
