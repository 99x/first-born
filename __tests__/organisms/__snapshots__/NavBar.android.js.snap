// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavBar with back button 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#486c86",
      "height": 56,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#486c86",
            "borderRadius": 0,
            "borderWidth": 0,
            "elevation": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#486c86",
                "fontSize": 20,
              },
              Object {
                "paddingRight": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "left",
            },
            Object {},
          ]
        }
      >
        Title
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders NavBar with custom button 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#486c86",
      "height": 56,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "left",
            },
            Object {},
          ]
        }
      >
        Title
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#486c86",
            "borderRadius": 0,
            "borderWidth": 0,
            "elevation": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#486c86",
                "fontSize": 20,
              },
              Object {
                "paddingRight": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders NavBar with drawer button 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#486c86",
      "height": 56,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#486c86",
            "borderRadius": 0,
            "borderWidth": 0,
            "elevation": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#486c86",
                "fontSize": 20,
              },
              Object {
                "paddingRight": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "left",
            },
            Object {},
          ]
        }
      >
        Title
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders NavBar with search button 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#486c86",
      "height": 56,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#486c86",
            "borderRadius": 0,
            "borderWidth": 0,
            "elevation": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#486c86",
                "fontSize": 20,
              },
              Object {
                "paddingRight": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "left",
            },
            Object {},
          ]
        }
      >
        Title
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders NavBar with title 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#486c86",
      "height": 56,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "left",
            },
            Object {},
          ]
        }
      >
        Title
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;
